File: src/main/java/application/history/history/presentation/interfaces/api/UpdateHistoryController.java
package application.history.history.presentation.interfaces.api;

import application.history.history.application.inputs.UpdateHistoryInput;
import application.history.history.application.services.UpdateHistoryContentService;
import org.springframework.format.annotation.DateTimeFormat;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import java.time.LocalDate;

@RestController
public class UpdateHistoryController {
    private final UpdateHistoryContentService service;

    public UpdateHistoryController(UpdateHistoryContentService updateHistoryContentService) {
        this.service = updateHistoryContentService;
    }

    @PutMapping("/histories/{date}")
    public ResponseEntity<Object> save(
        @PathVariable("date") @DateTimeFormat(pattern = "yyyy-MM-dd") LocalDate date,
        @RequestBody UpdateHistoryInput request
    ) {
        this.service.run(date, request);

        return ResponseEntity.noContent().build();
    }
}

File: src/main/kotlin/com/example/history/presentation/interfaces/api/UpdateHistoryController.kt
package com.example.history.presentation.interfaces.api

import com.example.history.application.inputs.UpdateHistoryInput
import com.example.history.application.services.FindHistoryUseCase
import com.example.history.application.services.SaveHistoryUseCase
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/histories")
class UpdateHistoryController(
    private val updateHistoryService: SaveHistoryUseCase,
    private val findHistoryService: FindHistoryUseCase
) {

    @PutMapping("/{id}")
    fun update(
        @PathVariable("id") id: String,
        @RequestBody request: UpdateHistoryInput
    ): ResponseEntity<Any> {
        val history = findHistoryService.findById(id);

        history.content = request.content;
        history.title = request.title;
        history.improvements = request.improvements;
        history.startDate = request.startDate;
        history.endDate = request.endDate;

        updateHistoryService.save(history)

        return ResponseEntity.noContent().build()
    }

}
